### PR TITLE
[atl] new port

### PR DESCRIPTION
**Pull request by yurybura**
This port ensures Visual Studio has ATL installed.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [ ] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [ ] The license declaration in `vcpkg.json` matches what upstream says.
- [ ] The installed as the "copyright" file matches what upstream says.
- [ ] The source code of the component installed comes from an authoritative source.
- [ ] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
Related: https://github.com/microsoft/vcpkg/pull/31604#issuecomment-1632495309

**Comment by dg0yt** [comment]
Possibly related: #24010

File: ports/atlmfc/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-  "description": "a stub package that ensures VS has ATL/MFC installed.",
-  "supports": "windows"
+  "port-version": 3,
+  "description": "A stub package that ensures Visual Studio has ATL/MFC installed.",
+  "supports": "windows",
+  "dependencies": [
+    {
+      "name": "atl",
+      "platform": "windows"
+    }
```
[the remark this answers is not in the document]
Comment (yurybura) [reply ports/atlmfc/vcpkg.json]: Done.

**Review by BillyONeal — APPROVED** [review]
While we think the experience of these kinds of 'system test' ports are suboptimal, in particular because versioning doesn't understand them and because binary caching causes them to do the wrong thing, we don't have concrete plans to introduce a better solution in the near term, so we're going to accept this.

Note that we are not going to bend over backwards in auditing the entire tree to ensure this is declared correctly, but we'll take changes that add such constraints where they are found.

**Pull request merged**